server/models.py

```diff
@@ -586,6 +595,31 @@ def revision(self, user_ids):
                       .order_by(Backup.created.desc())
                       .first())
 
+    def scores(self, user_ids, only_published=True):
+        """Return a list of Scores for this assignment and a group. Only the
+        maximum score for each kind is returned. If there is a tie, the more
+        recent score is preferred.
+        """
+        scores = Score.query.filter(
+            Score.user_id.in_(user_ids),
+            Score.assignment_id == self.id,
+            Score.archived == False,
+        ).order_by(Score.score.desc(), Score.created.desc())
+
+        scores_by_kind = {}
+        # keep only first score for each kind
+        for score in scores:
```

Comment:
The assignment detail page takes 4 seconds to load locally. The profiler is showing that it's spending a lot of time in the cursor. Calling `scores.all()` here doesn't seem to fix it - so it must be somewhere else.